File: warmup/gameofthrones1/gameofthrones1.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strings"
)

func main() {
	//file := os.Stdin
	file, _ := os.Open(`.\data\gameofthrones1.txt`)
	in, _ := ioutil.ReadAll(file)
	key := strings.TrimSpace(string(in))

	ans := "YES"
	chrCnts := make(map[rune]int)
	oddCnt := 0

	for _, chr := range key {
		chrCnts[chr]++
	}

	fmt.Fprintln(os.Stderr, chrCnts)

	if len(chrCnts) > 1 {
		for _, cnt := range chrCnts {
			if cnt%2 == 1 {
				oddCnt++
			}
			if oddCnt > 1 {
				ans = "NO"
				break
			}
		}
	}

	fmt.Println(ans)
}

File: warmup/gameofthrones1/gameofthrones1.py
from sys import stdin
#file = stdin
file = open( r".\data\gameofthrones1.txt" ) #stdin
data = file.read().strip()
file.close()

oddCnt = 0

for char in range(ord("a"),ord("z")+1):
    if (data.count(chr(char)) % 2 != 0):
        oddCnt += 1

if (oddCnt <= 1):
    print("YES")
else:
    print("NO")
